Counterexamples/VitaliSetNotMeasurable.lean

```diff
@@ -0,0 +1,318 @@
+  h.nullMeasurableSet
+
+lemma measurable_null_nullmeasurable {s t : Set ℝ}
+    (hm : NullMeasurableSet s) (hn : volume t = 0) : NullMeasurableSet (s ∪ t) :=
+  NullMeasurableSet.union_null hm hn
+
+lemma nullmeasurable_measurable_null {s : Set ℝ} (h : NullMeasurableSet s volume) :
+    ∃ t ⊆ s, MeasurableSet t ∧ volume t = volume s ∧ volume (s \ t) = 0 := by
+  have ⟨t, t_sub_s, t_m, t_eq_s⟩ := h.exists_measurable_subset_ae_eq
+  use t, t_sub_s, t_m
+  constructor
+  · exact measure_congr t_eq_s
+  · exact ae_le_set.mp t_eq_s.symm.le
+
+lemma shift_volume (s : Set ℝ) (c : ℝ) : volume ((fun x ↦ x + c) '' s) = volume s := by
+  simp only [image_add_right, measure_preimage_add_right]
+
+lemma shift_nullmeasurable {s : Set ℝ} (h : NullMeasurableSet s volume) (c : ℝ) :
+    NullMeasurableSet ((fun x ↦ x + c) '' s) volume := by
+  rcases nullmeasurable_measurable_null h with ⟨t, ts, tm, vs, vt⟩
+  rw [← union_diff_cancel ts, image_union]
+  refine measurable_null_nullmeasurable ?_ ?_
+  · have : MeasurableSet ((fun x ↦ x + c) '' t) := by
+      apply (MeasurableEmbedding.measurableSet_image ?_).mpr tm
+      exact measurableEmbedding_addRight c
+    exact measurable_nullmeasurable this
+  · rw [shift_volume (s \ t), vt]
+
+lemma union_volume_null {s t : Set ℝ} (hs : MeasurableSet s) (ht : volume t = 0) :
+    volume (s ∪ t) = volume s := by
+  have hu : s ∪ t = s ∪ (t \ s) := union_diff_self.symm
+  have hd : Disjoint s (t \ s) := disjoint_sdiff_right
+  have hz : volume (t \ s) = 0 := by
+    apply le_antisymm
+    · rw [← ht]
+      exact volume_mono diff_subset
+    · exact zero_le (volume (t \ s))
+  rw [hu, measure_union' hd hs, hz]
+  abel
+
+lemma biUnion_null {ι : Type*} {I : Set ι} {f : ι → Set ℝ}
+    (hs : I.Countable) : volume (⋃ i ∈ I, f i) = 0 ↔ ∀ i ∈ I, volume (f i) = 0 :=
+  measure_biUnion_null_iff hs
+
+lemma biUnion_volume {ι : Type*} {I : Set ι} {s : ι → Set ℝ}
+    (hc : I.Countable) (hd : I.PairwiseDisjoint s) (hm : ∀ i ∈ I, NullMeasurableSet (s i) volume) :
+    volume (⋃ i ∈ I, s i) = ∑' (i : ↑I), volume (s ↑i) := by
+  have : ∀ i ∈ I, ∃ t ⊆ s i, MeasurableSet t ∧ volume t = volume (s i) ∧ volume (s i \ t) = 0 := by
+    intro i i_I
+    exact nullmeasurable_measurable_null (hm i i_I)
+  choose! t t_s t_m t_v t_z using this
+  have h_st : ⋃ i ∈ I, s i = (⋃ i ∈ I, t i) ∪ (⋃ i ∈ I, (s i \ t i)) := by
+    -- This is probably not the most elegant proof of this rather trivial fact.
+    refine le_antisymm ?_ ?_
+    · intro x
+      simp only [mem_union, mem_iUnion₂]
+      rintro ⟨i, i_I, x_s⟩
+      rcases em (x ∈ t i) with x_t | x_nt
+      · left ; use i, i_I
+      · right ; use i, i_I
+        rw [mem_diff]
+        constructor <;> assumption
+    · refine union_subset ?_ ?_
+      · exact biUnion_mono (subset_refl I) t_s
+      · refine biUnion_mono (subset_refl I) ?_
+        intro i ?_
+        exact diff_subset
+  have hm_ti : ∀ i ∈ I, MeasurableSet (t i) := by
```

Comment:
I believe `hm_ti` is exactly `t_m`: so please just replace it in your proof below.